File: rotp.cpp
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int flipPancakes(string stack)
{
    int flips_made = 0;

    // Let's go through the stack
    for(int j=1;j< stack.length() ;j++)
    {
        // If current pancake is not like the one before, we will need to make a change
        if( stack[j] != stack[j-1] )
        {
            flips_made += 1;
        }
    }

    // If the last pancake is not "OK", we will need to make a change, otherwise return the flips made until now.
    if( stack[stack.length()-1] == '+')
    {
        return flips_made;
    } else {
        return flips_made + 1;
    }
}

int main()
{
    string line;
    int lines, flips;

    // Open file and check it's open
    try {

        ifstream file("sample.in");
        // Let's get the number of tests
        if( getline(file,line) )
        {
            lines = atoi(line.c_str());
            cout << "Lines: " << lines << endl;

            if(lines>0){
                for( int x=1; x<=lines; x++ )
                {
                    getline(file,line);
                    flips = flipPancakes(line);
                    cout << "Case [" << line  << "] #" << x << ": " << flips << endl;
                }
            }

        } else {
            cout << "Could not get the number of samples from the first line!" << std::endl;
        }

    } catch ( const ifstream::failure& e) {
        cout << "Could not open the samples file!" << std::endl;
    }
    return 0;
}

File: rotp.go
package main
import (
    "fmt"
	"strconv"
	"bufio"
	"os"
)

func check(e error) {
    if e != nil {
        panic(e)
    }
}

func flipStack(stack string) (int) {
	flips := 0

	// Loop through the stack, if the currect position is different from the last, flip it.
	for pos, char := range stack {
		if pos > 0 {
			if char != int32(stack[pos-1]) {
				flips += 1
			}
		}
	}

	// If the last pancake is not OK, flip it!
	if stack[ len(stack)-1 ] == '+' {
		return flips;
	} else {
		return flips + 1;
	}
}

func main() {
	file, err := os.Open("sample.in") 
    check(err)
	snl := bufio.NewScanner(file)

	// Read first line to get sample count
	snl.Scan()
	lines, err := strconv.Atoi( snl.Text() )
	fmt.Printf("Samples: %v\n", lines)
	check(err)

	// Get the count of samples, on per line and do the flipping
    for i:=1; i<=lines; i++ {
		snl.Scan()
		flips := flipStack( snl.Text() )
		fmt.Printf( "Case [%s] #%v: %v\n", snl.Text(), i, flips )
    }

	err = snl.Err()

}
